app/controllers/vm_common.rb

```diff
@@ -842,18 +842,18 @@ def parent_folder_id(vm)
     elsif vm.archived
       "xx-arch"
     elsif vm.cloud && vm.template
-      TreeBuilder.build_node_cid(vm.ems_id, 'ExtManagementSystem')
+      TreeBuilder.build_node_id(vm.ext_management_system)
     elsif vm.cloud && vm.availability_zone_id.nil?
-      TreeBuilder.build_node_cid(vm.ems_id, 'ExtManagementSystem')
+      TreeBuilder.build_node_id(vm.ext_management_system)
     elsif vm.cloud
-      TreeBuilder.build_node_cid(vm.availability_zone_id, 'AvailabilityZone')
+      TreeBuilder.build_node_id(AvailabilityZone.find_by(:id => vm.availability_zone_id))
```

Comment:
Why not do `vm.availability_zone` to make it more readable?

Reply:
@ZitaNemeckova because our backend doesn't provide this method :trollface:  cc @Ladas 
```
Vm.first.availability_zone
NoMethodError: undefined method `availability_zone' for #<ManageIQ::Providers::Vmware::InfraManager::Vm:0x000055d7007b84c8>
```

Reply:
But I agree, it's more readable :rofl:

Reply:
@skateman Really? I tried it on first one that I got for `Vm.find_by(:cloud => true)` and it worked. You tried it on Infra Vm. Maybe that's the problem :)